java/integTest/src/test/java/glide/SharedCommandTests.java

```diff
@@ -1055,6 +1057,105 @@ public void zcard(BaseClient client) {
         assertTrue(executionException.getCause() instanceof RequestException);
     }
 
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zrandmember(BaseClient client) {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0);
+        assertEquals(2, client.zadd(key1, membersScores).get());
+
+        String randMember = client.zrandmember(key1).get();
+        assertTrue(randMember.equals("one") || randMember.equals("two"));
+        assertNull(client.zrandmember("nonExistentKey").get());
+
+        // Key exists, but it is not a set
+        assertEquals(OK, client.set(key2, "bar").get());
+        ExecutionException executionException =
+                assertThrows(ExecutionException.class, () -> client.zrandmember(key2).get());
+        assertTrue(executionException.getCause() instanceof RequestException);
+    }
+
+    @SneakyThrows
+    @ParameterizedTest
+    @MethodSource("getClients")
+    public void zrandmemberWithCount(BaseClient client) {
+        String key1 = UUID.randomUUID().toString();
+        String key2 = UUID.randomUUID().toString();
+        Map<String, Double> membersScores = Map.of("one", 1.0, "two", 2.0);
+        assertEquals(2, client.zadd(key1, membersScores).get());
+
+        // Unique values are expected as count is positive
+        String[] randMembers = client.zrandmemberWithCount(key1, 4).get();
+        assertEquals(2, randMembers.length);
+        assertEquals(2, new HashSet<>(Arrays.asList(randMembers)).size());
+        assertTrue(
+                Arrays.stream(randMembers)
+                        .allMatch(member -> member.equals("one") || member.equals("two")));
+
+        // Duplicate values are expected as count is negative
+        randMembers = client.zrandmemberWithCount(key1, -4).get();
+        assertEquals(4, randMembers.length);
+        assertTrue(
+                Arrays.stream(randMembers)
+                        .allMatch(member -> member.equals("one") || member.equals("two")));
+
```

Comment:
Add a test with 0 as the second argument

Reply:
Did you mean 0 arguments, what is the point it returns an empty array?

Reply:
yes
or error - also good